src/services/refactors/moveToFile.ts

```diff
@@ -0,0 +1,1114 @@
+) {
+    const checker = program.getTypeChecker();
+    const prologueDirectives = takeWhile(oldFile.statements, isPrologueDirective);
+    if (oldFile.externalModuleIndicator === undefined && oldFile.commonJsModuleIndicator === undefined && usage.oldImportsNeededByNewFile.size === 0 && usage.newFileImportsFromOldFile.size === 0 && !newFileExists) {
+        deleteMovedStatements(oldFile, toMove.ranges, changes);
+        return [...prologueDirectives, ...toMove.all];
+    }
+
+    let useEsModuleSyntax = true;
+    if (newFileExists) {
+        const newFilePath = program.getSourceFile(newFile)?.path;
+        if (extensionFromPath(newFile) === Extension.Js && newFilePath && getImpliedNodeFormatForFile(newFilePath, /*packageJsonInfoCache*/ undefined, host, program.getCompilerOptions()) !== ModuleKind.ESNext && (oldFile.commonJsModuleIndicator || program.getCompilerOptions().verbatimModuleSyntax)) {
+            useEsModuleSyntax = false;
+        }
+    }
+    const quotePreference = getQuotePreference(oldFile, preferences);
+    const importsFromNewFile = createOldFileImportsFromNewFile(oldFile, usage.oldFileImportsFromNewFile, newFilename, program, host, useEsModuleSyntax, quotePreference);
+    if (importsFromNewFile) {
+        insertImports(changes, oldFile, importsFromNewFile, /*blankLineBetween*/ true, preferences);
+    }
+
+    deleteUnusedOldImports(oldFile, toMove.all, changes, usage.unusedImportsFromOldFile, checker);
+    deleteMovedStatements(oldFile, toMove.ranges, changes);
+    updateImportsInOtherFiles(changes, program, host, oldFile, usage.movedSymbols, newFilename, quotePreference);
+
+    const imports = getNewFileImportsAndAddExportInOldFile(oldFile, newFile, usage.oldImportsNeededByNewFile, usage.newFileImportsFromOldFile, changes, checker, program, host, useEsModuleSyntax, quotePreference, importAdder);
+    const body = addExports(oldFile, toMove.all, usage.oldFileImportsFromNewFile, useEsModuleSyntax);
+    if (newFileExists) {
+        const sourceFile = program.getSourceFile(newFile);
+        if (sourceFile && sourceFile.statements.length > 0) {
+            changes.insertNodesAfter(sourceFile, sourceFile.statements[sourceFile.statements.length - 1], body);
+        }
+        if (sourceFile && imports.length > 0) {
+            insertImports(changes, sourceFile, imports, /*blankLineBetween*/ true, preferences);
+        }
+    }
+    if (importAdder) {
+        importAdder.writeFixes(changes);
+    }
+    if (imports.length && body.length) {
+        return [
+            ...prologueDirectives,
+            ...imports,
+            SyntaxKind.NewLineTrivia as const,
+            ...body
+        ];
+    }
+
+    return [
+        ...prologueDirectives,
+        ...imports,
+        ...body,
+    ];
+}
+
+function getNewFileImportsAndAddExportInOldFile(

```

Comment:
Clarity nit:

Comments in this function would be helpful:

1. Why is `importAdder` sometimes undefined?
2. Why might it throw when using it, and what are we doing to recover?